### ISSUE TITLE
[Feature]: Zero-Configuration Rslib

### ISSUE THREAD

**Issue by colinaaa** [web-infra-dev/rslib#62]
### What problem does this feature solve?

Tools like [`pkgroll`](https://github.com/privatenumber/pkgroll) and [`microbundle`](https://github.com/developit/microbundle) offer zero-configuration support. It would be great if Rslib also supports using with zero-configuration.

### What does the proposed API look like?

Rslib can be run without `rslib.config.ts`. Most of the configuration can be read from `package.json`.

- Read `entry` from `main`, `module`, `umd` and `exports`
- Read `externals` from `dependencies`, `devDependencies` and `peerDependencies`
- Read `alias` from `package.json#imports`

Or, maybe create a new tool Rspkg that uses Rslib with zero-configuration.

**Comment by fi3ework** [comment]
Yeah! Using `package.json` as the single source of truth is in plan. Rslib also should check the correctness with it when providing a `rslib.config` file. We may support `autoExternal` at first (tracked in https://github.com/web-infra-dev/rslib/issues/46).

**Comment by spencerbeggs** [comment]
I have an abstraction library that does this: https://github.com/savvy-web/rslib-builder

The transformation is an RSBuild plugin. My library is a bit custom, but you can pull out that code from here: https://github.com/savvy-web/rslib-builder/blob/main/package/src/rslib/plugins/auto-entry-plugin.ts
